cascade-cli/src/Cascade/CLI.hs

```diff
@@ -13,92 +13,40 @@ Portability : POSIX
  where
   connectionInfo =
     Postgres.ConnectInfo { connectHost = host, connectPort = port, connectUser = user, connectPassword = password, connectDatabase = database }
   acquire = Postgres.connect connectionInfo
 
-runCascadeApi :: Options -> IO ()
-runCascadeApi Options {..} = do
-  databaseConnectionPool <- mkDatabaseConnectionPool postgresOptions
-  Cascade.Api.main Cascade.Api.Config { port = httpPort, withDatabaseConnection = Pool.withResource databaseConnectionPool }
-
-cascadeVersion :: String
-cascadeVersion = intercalate "\n" [cVersion, cHash, cDate]
- where
-  cVersion, cHash, cDate :: String
-  cVersion = "Cascade CLI v" <> showVersion Meta.version
-  cHash    = "Git revision: " <> $(gitHash)
-  cDate    = "Last commit:  " <> $(gitCommitDate)
-
-versionP :: Parser (a -> a)
-versionP = infoOption cascadeVersion <| mconcat [long "version", short 'v', help "Show cascade's version"]
-
-data PostgresOptions = PostgresOptions
-  { host     :: String
-  , port     :: Word16
-  , user     :: String
-  , password :: String
-  , database :: String
-  }
-
-data Options = Options
-  { httpPort        :: Int
-  , postgresOptions :: PostgresOptions
-  }
-
-postgresOptionsP :: Parser PostgresOptions
-postgresOptionsP =
-  PostgresOptions
-    <$> strOption (mconcat [long "postgres-host", metavar "CASCADE_POSTGRES_HOST", value "localhost", showDefault, help "PostgreSQL host"])
-    <*> option auto (mconcat [long "postgres-port", metavar "CASCADE_POSTGRES_PORT", value 5432, showDefault, help "PostgresSQL port"])
-    <*> strOption (mconcat [long "postgres-user", metavar "CASCADE_POSTGRES_USER", value "cascade", showDefault, help "PostgreSQL user"])
-    <*> strOption (mconcat [long "postgres-password", metavar "CASCADE_POSTGRES_PASSWORD", value "", showDefault, help "PostgreSQL passowrd"])
-    <*> strOption
-          (mconcat [long "postgres-database", metavar "CASCADE_POSTGRES_DATABASE", value "cascade-api", showDefault, help "PostgreSQL database"]
-          )
-
-optionsP :: Parser Options
-optionsP =
-  Options
-    <$> option auto (mconcat [long "http-port", metavar "CASCADE_HTTP_PORT", value 3141, showDefault, help "Port number of Cascade Api"])
-    <*> postgresOptionsP
+runCascadeApi :: Config.Final -> IO ()
+runCascadeApi Config {..} = do
+  databaseConnectionPool <- mkDatabaseConnectionPool postgresConfig
+  Cascade.Api.main Cascade.Api.Config { port = HttpPort.un httpPort, withDatabaseConnection = Pool.withResource databaseConnectionPool }
 
-cliP :: ParserInfo Options
-cliP = info (helper <*> versionP <*> optionsP) <| fullDesc <> progDesc "Cascade Cli"
+getFinalConfig :: IO (Validation (Validation.Errors Config.Partial Config.Final) Config.Final)
+getFinalConfig = join <| Config.finalize . fold <$> sequence [Environment.readConfig, Options.readConfig]
 
 main :: IO ()
-main = execParser cliP >>= runCascadeApi
+main = do
+  vConfig <- getFinalConfig
+  case vConfig of
+    Validation.Failure e -> print e
```

Comment:
Let's just print on stdout that something wrong happend. We will soon pretty print the errors.